### ISSUE TITLE
Add validate subcommand

### ISSUE THREAD

**Issue by jelovirt** [dita-ot/dita-ot#4397]
## Description
Add dedicated subcommand for validation. The process will run pre-processing to discover error or warnings. Instead of producing output, it would only log issues and return with non-zero exit code.

There will be a new extension point that plug-ins can use to run custom Ant targets for additional validation. If there are multiple plugins that use the extension point, then multiple validation steps are run in _some_ order. Each should just log warnings and errors, not product any output.

The validation subcommand could also produce a list of all error and warnings in some format. However, you can output the normal log into an XML file or using a custom build listener you can generate any output you want.

### Example uses

```shell
$ dita validate -i userguide.ditamap
$ dita validate -i userguide.ditamap --processing-mode=strict
```

## Open items

- [ ] Should the default processing mode be strict or lax? For conversion the default is lax.

**Comment by jelovirt** [comment]
My initial preference would be to use the same processing mode for validation as for conversion. If you want strict processing, it's easy to switch it on. However, for validation it feels the default should maybe be strict 🤷🏽

**Comment by fwegmann** [comment]
I would agree: If you use DITA-OT for validation purposes, then there is the express intention of checking sources thoroughly. So that should indeed be the default.

**Comment by chrispy-snps** [comment]
My first thought was also that validation should default to the same behavior as transformation, just a "read-only" version of it. But then I realized that from a user perspective, having a validator default to "relaxed validation" is counter-intuitive. So I also agree that `validate` should default to `strict`.

**Comment by stefan-jung** [comment]
@jelovirt I really appreciate this new feature. This is a massive improvement. I developed something similar, basically by setting a property to exit processing after preprocessing. It would be awesome to validate in parallel also with stylistic rules as implemented in Schematron (for DITA) or Vale (for DITA and Markdown). In an ideal world I can then use this validate function to validate the root DITA-OT project file and can define a quality gate which specifies if the code is syntactically but also stylistically correct.

**Comment by chrispy-snps** [comment]
@stefan-jung - having `validate` somehow include Schematron/Vale in project-file-scoped validation would be awesome.
